### PR TITLE
adjust documentation to be symfony 3 compatible

### PR DESCRIPTION
**Pull request by dbu**
symfony 3 does not have a ContainerAware to extend. but anyway that is bad practice and not needed in this case.

continue from #298

### CODE REVIEW

File: Resources/doc/index.rst [hunk 1]
```diff
@@ -124,11 +124,18 @@ An example builder class would look like this:
...
+    use Doctrine\Common\Persistence\ManagerRegistry;
     use Knp\Menu\FactoryInterface;
-    use Symfony\Component\DependencyInjection\ContainerAware;
 
-    class Builder extends ContainerAware
+    class Builder
     {
+        private $registry;
+        
+        public function __construct(ManagerRegistry $registry)
```
Comment (stof) [inline Resources/doc/index.rst]: This is wrong. This example is about documenting menus rendered as ``knp_menu_render('AppBundle:Builder:mainMenu')``, so you cannot inject the ManagerRegistry in the constructor. This should be using ContainerAwareInterface here.

Registering builders as services is documented in a different chapter

**Comment by JHGitty** [comment]
@stof Sorry, deleted my text already - your answer was too fast. Thanks! :+1:

**Comment by dbu** [comment]
oh, right you are stof. updated it. better?

**Comment by BallisticPain** [comment]
Gents,

I thought the ContainerAwareInterface was being deprecated as well? Maybe I'm mistaken.

I was just looking to work on this based on the Issue #290 
So I believe this fixes #290 pending my above question.

Thoughts?

**Comment by JHGitty** [comment]
@BallisticPain The ``ContainerAwareInterface`` is **not** deprecated because it is useful in combination with ``ContainerAwareTrait``.

**Comment by stof** [comment]
@JHGitty actually more the opposite: ContainerAwareTrait is useful when you need to implement ContainerAwareInterface :smile:

**Comment by JHGitty** [comment]
@stof Yes, thanks. :-)

**Pull request merged**